Graham.cpp

```diff
@@ -1,87 +1,220 @@
 }
 
-double getSlope(p1, p2)
+// Finds the distance of two points
+int Graham::distance(Point P1, Point P2)
+{
+    int x_Distance = P1.x - P2.x;
+    int y_Distance = P1.y - P2.y;
+    return (x_Distance * x_Distance + y_Distance * y_Distance);
+}
+
+
+std::vector<Point> Graham::Graham_Scan(std::vector<Point> pointSet, std::vector<Point> hull)
 {
-    if p1[0] == p2[0]
+
+    int i, j , k = 0;
+    int top = 2;
+
+    Point tmp;  // temporary point
+    int n = pointSet.size();
+
+    // Less than 3 points
+    if (pointSet.size() < 3)
+    {
+        return hull;
+    }
+
+    // Finds the lowest point (Y) and left-most (X)
+    for(int i = 1; i < n; i++)
     {
-        // Don't return 1 or 0 --> They are slopes
-        // Return infinity
-        return std::numeric_limits<double>::infinity();
+        if(pointSet[i].y < pointSet[k].y || pointSet[i].y == pointSet[k].y && pointSet[i].x < pointSet[k].x)
+        {
+            // Set k to minimum point
+            k = i;
+        }
+    }
+
+    // Set temporary point to the first point
+    tmp = pointSet[0];
+
+    // Swap places
+    pointSet[0] = pointSet[k];
+    pointSet[k] = tmp;
+
+    // Sort from smallest to largest polar angle
+    for(i = 1; i < n-1; i++)
+    {
+        k = i;
+
+        for(j = i+1; j < n; j++)
+        {
+            int angle1 = polarAngle(pointSet[j], pointSet[k], pointSet[0]);
+            int angle2 = polarAngle(pointSet[j], pointSet[k], pointSet[0]);
+            // Current polar angle > the previous polar angle
+            if(angle1 > 0  || angle2 == 0 && distance(pointSet[0], pointSet[j]) < distance(pointSet[0], pointSet[k]))
+            {
+                // k-> polar angle is stored in the minimum point
+                // or the same distance from the nearest origin
+                k = j;
+            }
+        }
+
+        // Swap points again
+        tmp = pointSet[i];
+        pointSet[i] = pointSet[k];
+        pointSet[k] = tmp;
     }
-    else
+
+
+    // Push the three points into hull
+    hull.push_back(pointSet[0]);
+    hull.push_back(pointSet[1]);
+    hull.push_back(pointSet[2]);
+
+    // Find the rest of the points n-3
+    for(i = 3; i < n; i++)
     {
-        return 1.0 * (p1[1] - p2[1]) / (p1[0]-p2[0]);
+        // Not left-turn
+        while(polarAngle(pointSet[i], hull[top], hull[top-1]) >= 0)
+        {
+            top--;
+            if(polarAngle(pointSet[i], hull[top], hull[top-1]) <  0)
+            {
+                disregardSet.push_back(pointSet[i]);
+            }
+            hull.pop_back();
+
+        }
+
+        // Point connecting the current point to the all points
+        hull.push_back(pointSet[i]);
+
+        ++top;
+
     }
-    // Compare slope in unit circle
+    // for(int i = 0; i < disregardSet.size(); i++)
+    // {
+    //     std::cout << "X:" << disregardSet[i].x << "\tY:" << disregardSet[i].y << std::endl;
+    // }
+    return hull;
+}
+
+
+int main(int argc, char* argv[])
```

Comment:
Change this from being a main function to some other function accessible from other files.